bsmetadata/metadata_processors.py

```diff
@@ -38,11 +39,13 @@ def process_global(self, metadata_attrs: Dict[str, Any]) -> Optional[str]:
         prefix will be "url: wikipedia.com/Apple".
 
         Args:
-            metadata_attrs: All attributes of this metadata instance. Each global metadata instance is expected to have an attribute "key"
-            of type string and a corresponding "value" of arbitrary type.
+            metadata_attrs (:obj:`Dict[str, Any]): All attributes of this metadata instance.
```

Comment:
Docstring changes like this are just adding types like other HuggingFace repos do.

Reply:
I'm not an expert on type hinting in python, but shouldn't it be sufficient that we already have the types in the method signature (e.g., `def process_global(self, metadata_attrs: Dict[str, Any]) -> Optional[str]:`)? I would think that storing this information in the signature **and** the docstring makes things more error prone as it's likely that someone changing the method signature will forget to adapt it in one of the two places.

Reply:
Oh and just a nitpick, there's a single `` ` `` missing after `` `Dict[str, Any]``

Reply:
>  storing this information in the signature and the docstring makes things more error prone

Indeed. I wasn't really considering the side effect but merely applying the practice of HuggingFace repos.


> there's a single ` missing after `Dict[str, Any]

A case in point 😅

(I'm gonna alter the history to remove docstring changes.)